From bbed15149f4aa6152cd466fbf6d25e23fb2834fc Mon Sep 17 00:00:00 2001
From: aaronw <aaron.williams@cavium.com>
Date: Mon, 4 Mar 2019 21:25:22 -0800
Subject: [PATCH 0888/1239] octeontx: octeontx2: use strlcpy instead of strncpy

strncpy does not terminate strings properly so this might be a
possible security hole.  Changed to strlcpy to fix this.

Change-Id: I60f942bbf9cc8ac5a41aadd50e48c7a4ef68c5f7
Signed-off-by: aaronw <aaron.williams@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/5421
Reviewed-by: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Tested-by: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
---
 board/Marvell/octeontx/octeontx.c   | 2 +-
 board/Marvell/octeontx2/octeontx2.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/Marvell/octeontx/octeontx.c b/board/Marvell/octeontx/octeontx.c
index 303497887c..aa346f55c3 100644
--- a/board/Marvell/octeontx/octeontx.c
+++ b/board/Marvell/octeontx/octeontx.c
@@ -122,7 +122,7 @@ void octeontx_parse_board_info(void)
 	str = fdt_getprop(gd->fdt_blob, node, "BOARD-MODEL", &len);
 	debug("fdt: BOARD-MODEL str %s len %d\n", str, len);
 	if (str) {
-		strncpy(g_cavm_bdt.type, str, sizeof(g_cavm_bdt.type));
+		strlcpy(g_cavm_bdt.type, str, sizeof(g_cavm_bdt.type));
 		debug("fdt: BOARD-MODEL bdt.type %s \n", g_cavm_bdt.type);
 	} else {
 		printf("Error: cannot retrieve board type from fdt\n");
diff --git a/board/Marvell/octeontx2/octeontx2.c b/board/Marvell/octeontx2/octeontx2.c
index 20ed369bae..d76d2b044d 100644
--- a/board/Marvell/octeontx2/octeontx2.c
+++ b/board/Marvell/octeontx2/octeontx2.c
@@ -104,7 +104,7 @@ void octeontx_parse_board_info(void)
 	str = fdt_getprop(gd->fdt_blob, node, "BOARD-MODEL", &len);
 	debug("fdt: BOARD-MODEL str %s len %d\n", str, len);
 	if (str) {
-		strncpy(g_cavm_bdt.type, str, sizeof(g_cavm_bdt.type));
+		strlcpy(g_cavm_bdt.type, str, sizeof(g_cavm_bdt.type));
 		debug("fdt: BOARD-MODEL bdt.type %s \n", g_cavm_bdt.type);
 	} else {
 		printf("Error: cannot retrieve board type from fdt\n");
-- 
2.29.0

